src/mpid/ch4/shm/posix/release_gather/release_gather.h

```diff
@@ -19,7 +19,8 @@ extern MPIDI_POSIX_release_gather_tree_type_t MPIDI_POSIX_Bcast_tree_type,
         while (MPL_atomic_acquire_load_uint64(ptr) < (value))    { \
             if (++spin_count >= 10000) {                           \
                 /* Call progress only after waiting for a while */ \
-                MPID_Progress_test();                              \
+                int err_ = MPID_Progress_test();                   \
+                MPIR_Assert(err_ == MPI_SUCCESS);                  \
```

Comment:
Just want to be clarified, why here is not using `error_check`?

Reply:
Because I am not sure whether the function calling this macro has the `fn_fail` label or whether it needs return error.